Vignettes/howtoaddvariables.Rmd

```diff
@@ -188,87 +188,45 @@ CustomFunctionName <- function(Vars from variableStart following same order){
 
 Pack-years is a complex derived variable often used by researchers to quantify the amount of cigarette use over a period of time. This derived variable incorporates numerous CCHS smoking variables, along with age.
 
-### Step 1a. Developing shorthand names for derived variable function
-
-When creating a derived variable like pack-years that uses many complex variables, it might be easier to create shorthand names you can be reference when creating the function. 
-
-```{r}
-#Smoking variables
-#TypeOfSmoker <- SMKDSTY 
-#Age_cont <- DHHGAGE_cont
-#stpd <- SMK_09A_B #former daily (quit <3 years ago)
-#stpdy <- SMKG09C #former daily (quit >=3 years ago)
-#agec1 <- SMKG01C_cont #age smoked first cigarette
-#agecigd <- SMKG203_cont #age started smoking daily
-#agecigfd <- SMKG207_cont #age started smoking daily (former daily)
-#cigdayd <- SMK_204 # number of cigarettes smoked per day (daily smoker)
-#cigdayo <- SMK_05B # number of cigarettes smoked per day (occasional smoker)
-#cigdayf <- SMK_208 # number of cigarettes smoked per day (former daily)
-#dayocc <- SMK_05C # number of days smoked at least one cigarette
-#s100 <- SMK_01A # smoked 100 cigarettes in lifetime (y/n)
-```
-
-**NOTE:** R will **not** read the above transformations. However, the shorthand names can be used in the custom function **if** they are listed in the same order as how their corresponding CCHS variables are listed in `variableDetails.csv`.
-
-For example, let's say you want create a derived variable using age as a continuous variable, height and weight. Shorthand names for the 3 variables can look like this:
-
-```{r}
-#Age_cont <- DHHGAGE_cont
-#Height <- HWTGHTM
-#Weight <- HWTGWTK
-```
-
-In the variableStart column of `variableDetails.csv`, the CCHS variables will be listed in this order:
-
-`DerivedVar::[DHHGAGE_cont, HWTGHTM, HWTGWTK]`
-
-With your custom function following the same order as the variableStart column, you are able to use the shorthand names in your function
-
-```
-derivedFunction <- function(Age_cont, Height, Weight) {
-}
-```
-
-Shorthand names is way to keep track of variables and not run into errors when creating a derived variable.
-
-### Step 1b. Creating derived variable function
+### Step 1. Creating a derived function
 
 With complex derived variables, sometimes it is necessary to create functions within the custom function. For pack-years, a nested function was used to create an intermediate smoking variable that was used in the main function.
 
 ```{r, warning=FALSE}
-#Smoking pack-years
 PackYears_fun <-
-  function(TypeOfSmoker, Age_cont, stpd, stpdy, agecigd, agecigfd, cigdayd, cigdayo, cigdayf, dayocc, agec1, s100) {
+  function(SMKDSTY, DHHGAGE_cont, SMK_09A_B, SMKG09C, SMKG203_cont, SMKG207_cont, SMK_204, SMK_05B, SMK_208, SMK_05C, SMKG01C_cont, SMK_01A) {
```

Comment:
Since we have these functions as part of cchsflow package we can load it instead of writing all the functions out.
@Doug unless keeping them in vignette is beneficial to user understanding.

Reply:
This is an example to show users how to create a derived variable. I think it would help to show users a complex example like packyears. 

@DougManuel what do you think?

Reply:
I definitely think we want to have an example. Smoking is a bit intimidating. Why don't you put a caveat along the lines, 
"even complex derived variables such packYears_fun can be calculated.

Only, is the name `PackYears_fun` or `pack-years_der` or `pack_years_der`?